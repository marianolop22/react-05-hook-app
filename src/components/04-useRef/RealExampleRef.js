import React, { useState } from 'react';
import '../02-useEffect/effects.css';
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Real Example ref</h1>
            <hr/>


            { show && <MultipleCustomHooks/>}

            <button
                className="btn btn-info mt-5"
                onClick={ ()=> {
                    setShow (!show);
                }}
            >
                Show/Hide
            </button>

        </div>
    )
}
