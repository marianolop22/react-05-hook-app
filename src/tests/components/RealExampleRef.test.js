import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from "../../components/04-useRef/RealExampleRef";

describe ('Pruebas de RealExampleRef', ()=> {
    
    const wrapper = shallow(<RealExampleRef/>);

    test('debe mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe mostrar el MultipleCustomHooks', () => {

        // const wrapper = shallow(<RealExampleRef/>);
        wrapper.find('button').at(0).simulate('click');

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
        
    });

});