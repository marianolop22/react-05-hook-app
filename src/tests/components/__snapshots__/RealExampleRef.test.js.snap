// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de RealExampleRef debe mostrar el MultipleCustomHooks 1`] = `
<div>
  <h1>
    Real Example ref
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-info mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Pruebas de RealExampleRef debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example ref
  </h1>
  <hr />
  <button
    className="btn btn-info mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
